### PR TITLE
Fix duplicate ickle results

### PR DESCRIPTION
**Pull request by varunshastry**

### CODE REVIEW

**Review by jasperRob — APPROVED** [review]

File: kogitoq/kogito-common/src/main/java/life/genny/kogito/common/service/NavigationService.java [hunk 1]
```diff
@@ -195,9 +196,9 @@ public String getRedirectCodeByUser() {
 		String defCode = "";
 		try {
 			BaseEntity user = beUtils.getUserBaseEntity();
-			List<EntityAttribute> priIsAttributes = beaUtils.getBaseEntityAttributesForBaseEntityWithAttributeCodePrefix(user.getRealm(), user.getCode(), PRI_IS_PREFIX);
+			Set<EntityAttribute> priIsAttributes = beaUtils.getBaseEntityAttributesForBaseEntityWithAttributeCodePrefix(user.getRealm(), user.getCode(), PRI_IS_PREFIX);
 			if (priIsAttributes.size() > 0) {
-				EntityAttribute attr = priIsAttributes.get(0);
+				EntityAttribute attr = priIsAttributes.iterator().next();
```
Comment (Expressionless) [inline kogitoq/kogito-common/src/main/java/life/genny/kogito/common/service/NavigationService.java:201]: Can we move away from PRI_IS please?

Comment (Expressionless) [reply kogitoq/kogito-common/src/main/java/life/genny/kogito/common/service/NavigationService.java:201]: Should be using LNK_DEF

File: qwandaq/src/main/java/life/genny/qwandaq/serialization/attribute/Attribute.java [hunk 2]
```diff
@@ -1,25 +1,22 @@
...
 /*
  * A representation of BaseEntity in the cache
  * 
  * @author Varun Shastry
  */
 public class Attribute implements CoreEntitySerializable {
 	
-	private String dtype;
-	
-	private Long id;
```
Comment (Expressionless) [inline qwandaq/src/main/java/life/genny/qwandaq/serialization/attribute/Attribute.java:21]: Not ready to remove Id yet (no fyodor modifications)

Comment (varunshastry) [reply qwandaq/src/main/java/life/genny/qwandaq/serialization/attribute/Attribute.java:21]: Fixed

File: kogitoq/kogito-common/src/main/java/life/genny/kogito/common/service/NavigationService.java (hunk 1 above)
Comment (varunshastry) [reply kogitoq/kogito-common/src/main/java/life/genny/kogito/common/service/NavigationService.java:201]: To be addressed as part of a dedicated Jira ticket.

File: qwandaq/src/main/java/life/genny/qwandaq/validation/Validation.java [hunk 3]
```diff
@@ -296,4 +297,18 @@ public CoreEntitySerializable toSerializableCoreEntity() {
 		validation.setSelectionGroup(validationListConverter.convertToString(getSelectionBaseEntityGroupList()));
 		return validation;
 	}
+
+	@Override
+	public int hashCode() {
+		HashCodeBuilder hcb = new HashCodeBuilder();
+		hcb.append(getRealm());
+		hcb.append(getCode());
+		return hcb.toHashCode();
```
Comment (Expressionless) [inline qwandaq/src/main/java/life/genny/qwandaq/validation/Validation.java]: This method is shadowing the CodedEntity@hashCode method which has the exact same implementation, so this method can be removed all together

**Review by Expressionless — CHANGES_REQUESTED** [review]
see notes on Validation.java

File: qwandaq/src/main/java/life/genny/qwandaq/validation/Validation.java (hunk 3 above)
Comment (varunshastry) [reply qwandaq/src/main/java/life/genny/qwandaq/validation/Validation.java]: Fixed

**Review by weslyvo81 — APPROVED** [review]

**Review by Expressionless — APPROVED** [review]
lgtm

**Pull request merged**